src/snc/agents/hedgehog/policies/big_step_base_policy.py

```diff
@@ -102,7 +102,7 @@ def get_index_all_forbidden_activities(self, nonidling_res) -> Set[int]:
                 self.buffer_processing_matrix,
                 self.constituency_matrix
             )
-            if len(fa_s) > 0:
+            if fa_s:
```

Comment:
Seeing this change, we might improve this method  further:
First we initialise `ind_forbidden_activities` as a `set`:
```
ind_forbidden_activities = set()
```
Then, we avoid the `if` and use directly `union`:
```
ind_forbidden_activities.union(set(fa_s))
```
Finally, there won't be needed to transform the list into a set anymore.

But we can leave it for another PR if you prefer.

Reply:
I'll leave it for now. Keen to merge this in before I change Laptops.